policy/reviewer/reviewer.go

```diff
@@ -73,96 +73,130 @@ func selectRandomUsers(n int, users []string, r *rand.Rand) []string {
+	var usersToRequest []string
 	pendingLeafNodes := findLeafChildren(result)
-	var requestedUsers []string
-
-	logger.Debug().Msgf("Collecting reviewers for %d pending leaf nodes", len(pendingLeafNodes))
 
 	for _, child := range pendingLeafNodes {
 		allUsers := make(map[string]struct{})
-		shoveIntoMap(allUsers, child.ReviewRequestRule.Users)
+
+		for _, user := range child.ReviewRequestRule.Users {
+			allUsers[user] = struct{}{}
+		}
 
 		if len(child.ReviewRequestRule.Teams) > 0 {
-			teamMembers, err := selectTeamMembers(prctx, child.ReviewRequestRule.Teams, r)
+			teamMembers, err := selectTeamMembers(prctx, child.ReviewRequestRule.Teams)
 			if err != nil {
-				logger.Warn().Err(err).Msgf("Unable to get member listing for teams, skipping team member selection")
+				logger.Warn().Err(err).Msgf("failed to get member listing for teams, skipping team member selection")
+			}
+			for _, user := range teamMembers {
+				allUsers[user] = struct{}{}
 			}
-			shoveIntoMap(allUsers, teamMembers)
 		}
 
 		if len(child.ReviewRequestRule.Organizations) > 0 {
-			orgMembers, err := selectOrgMembers(prctx, child.ReviewRequestRule.Organizations, r)
+			orgMembers, err := selectOrgMembers(prctx, child.ReviewRequestRule.Organizations)
 			if err != nil {
-				logger.Warn().Err(err).Msg("Unable to get member listing for org, skipping org member selection")
+				logger.Warn().Err(err).Msg("failed to get member listing for org, skipping org member selection")
+			}
+			for _, user := range orgMembers {
+				allUsers[user] = struct{}{}
 			}
-			shoveIntoMap(allUsers, orgMembers)
 		}
 
-		allCollaboratorPermissions, err := prctx.RepositoryCollaborators()
+		collaboratorsToConsider, err := prctx.RepositoryCollaborators()
 		if err != nil {
-			return nil, errors.Wrap(err, "Unable to list repository collaborators")
+			return nil, errors.Wrap(err, "failed to list repository collaborators")
 		}
 
-		if child.ReviewRequestRule.Admins {
-			var repoAdmins []string
-			for user, perm := range allCollaboratorPermissions {
-				if perm == common.GithubAdminPermission {
-					repoAdmins = append(repoAdmins, user)
+		if child.ReviewRequestRule.WriteCollaborators {
+			for user, perm := range collaboratorsToConsider {
+				if perm == common.GithubWritePermission {
+					allUsers[user] = struct{}{}
 				}
 			}
-			shoveIntoMap(allUsers, repoAdmins)
 		}
 
-		if child.ReviewRequestRule.WriteCollaborators {
-			var repoCollaborators []string
-			for user, perm := range allCollaboratorPermissions {
-				if perm == common.GithubWritePermission {
-					repoCollaborators = append(repoCollaborators, user)
-				}
+		if child.ReviewRequestRule.Admins {
+			logger.Debug().Msg("Selecting admins for review")
+			admins, err := selectAdmins(prctx)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to select admins")
+			}
+
+			for _, admin := range admins {
+				allUsers[admin] = struct{}{}
+				collaboratorsToConsider[admin] = common.GithubAdminPermission
```

Comment:
I thought these users would already be in `collaboratorsToConsider` as part of listing all collaborators. Is that not true?

Reply:
you're correct, I missed this in the last commit.